src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,60 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Class<?>> interfaceClass = new HashMap<>();
+
+    static {
+        interfaceClass.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceClass.put(ProductParser.class, ProductParserImpl.class);
+        interfaceClass.put(ProductService.class, ProductServiceImpl.class);
+    }
+
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public <T> T getInstance(Class<T> interfaceClazz) {
+        if (!interfaceClazz.isAnnotationPresent(Component.class)) {
```

Comment:
The check for the @Component annotation should be on the implementation class, not the interface class. This is because the implementation class should be annotated with @Component.